### PR TITLE
doc(tests): Add ToC, reorder content, adjust content

### PR DESCRIPTION
**Pull request by gyptazy**
**How to categorize this PR?**

/kind enhancement
/area os
/os garden-linux

**What this PR does / why we need it**:
 * Adding ToC to tests documentation
 * Adding content for `Misc`
 * Reordering content
 * Fix wording

**Release note**:
NONE

```feature user

```

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -210,7 +330,68 @@ Run the tests (be sure you properly mounted the Garden Linux repository to the c
...
+- mount build result directory to `/build` (if not part of the Garden Linux file tree)
+- mount directory with configfile to `/config`
+
+```
+docker run --cap-add SYS_ADMIN --security-opt apparmor=unconfined -it --rm -v `pwd`:/gardenlinux gardenlinux/integration-test:dev /bin/bash
+```
+
+Run the tests (be sure you properly mounted the Garden Linux repository to the container and you are in `/gardenlinux/tests`):
+
+    pytest --iaas=chroot --configfile=/config/mychrootmconfig.yaml integration/
```
Comment (Vincinator) [inline tests/README.md:391]: pedantic remark: I think the `m`  in mychroot**m**config.yaml is a typo? 
not related to to the PR, but while we are at it?

Comment (Vincinator) [inline tests/README.md:391]: pedantic remark: I think the `m`  in mychroot**m**config.yaml is a typo? 
not related to to the PR, but while we are at it?

**Comment by Vincinator** [comment]
besides small remarks above, looks good to me

File: tests/README.md [hunk 2]
```diff
@@ -23,21 +42,122 @@
...
+## Prerequisites
 
 Build the integration test container with all necessary dependencies. This container image will contain all necessary Python modules as well as the command line utilities by the Cloud providers (i.e. AWS, Azure and GCP) and KVM/QEMU.
 
     make --directory=docker build-integration-test
 
 The resulting container image will be tagged as `gardenlinux/integration-test:<version>` with `<version>` being the version that is returned by `bin/garden-version` in this repository. All fruther tests run inside this container.
 
-## 2. Tests by Cloud Provider
+## Examples for supported Cloud Provider
```
[the remark this answers is not in the document]
Comment (gyptazy) [reply tests/README.md:53]: Thanks, that fits better

File: tests/README.md (hunk 1 above)
Comment (gyptazy) [reply tests/README.md:391]: Thanks, fixed.

**Review by Vincinator — APPROVED** [review]
thanks, looks good

**Pull request merged**